### PR TITLE
Create standalone versions of story and science levels

### PR DESCRIPTION
**Pull request by molly-moen**
Note: this PR looks like it is a lot of lines to review, but most of those lines are the new `.level` files.

Some CSC levels use block pools and helper libraries that are not available in standalone Sprite Lab. This PR makes two new standalone projects that are sub-types of Sprite Lab, Story and Science. Story includes the new book covers pool, and Science contains the new particles pool. The levels themselves will probably need some editing by levelbuilders to get the right blocks in the toolbox, for now they are copies of sprite lab with a couple new blocks from the new pools. But adding them here enables levelbuilders to edit them.

I followed a similar pattern to [poetry](https://github.com/code-dot-org/code-dot-org/pull/42863) to make these new sub-types. With this change, levelbuilders can specify the sub-type for a sprite lab level. The default is Sprite Lab.
Screenshot 2023-01-27 at 3 41 58 PM

As of right now, the only levels with the sub-type Science or Story are the standalone projects themselves. If you create one of these projects (only doable via a url) or remix one, you get the appropriate naming in the projects table, and the url will be `/projects/story/...` or `/projects/science/...` instead of `/projects/spritelab/...`
Screenshot 2023-01-27 at 3 33 26 PM

Sprite Lab share/remix for non-story or science levels is unchanged.

## Links

- [spec](https://docs.google.com/document/d/1i8uzZ9D4KkeItPJf3_mLsN0UKIX87tZzu7uOn4lDgg4/edit#heading=h.b291g56tcxkw)
- jira ticket: [SL-519](https://codedotorg.atlassian.net/browse/SL-519)

## Testing story
Tested locally that you can now share/remix story and science projects, and Sprite Lab share/remix is unchanged.

## Follow-up work
- Improve the standalone project levels
- Get exemplar links working [ticket](https://codedotorg.atlassian.net/browse/SL-531)
- Work with the CSC team to set the sub-type on the appropriate Story and Science levels.

### CODE REVIEW

File: dashboard/app/views/levels/editors/fields/_special_level_types.html.haml [hunk 1]
```diff
@@ -55,3 +55,9 @@
         = text_field_tag 'level[contained_level_names][]', level_name, placeholder: 'name of contained level'
       #plusAnswerContainedLevel
         %i.fa.fa-plus-circle
+
+  - if (@level.is_a?(GamelabJr))
+    .field
+      = f.label :standalone_app_name, 'Level Subtype'
+      %p Select the subtype for the level. This will determine which blocks are available in the share/remix version of the level.
```
Comment (mikeharv) [inline dashboard/app/views/levels/editors/fields/_special_level_types.html.haml:62]: Nit: Should we add a note that engineers need to add additional subtypes?

**Review by mikeharv — APPROVED** [review]
The code changes here all look good! I pulled and tested, including:
* Remixing a CSC level on staging (into a broken state), then fixing the project on this branch by changing `spritelab` to `science` in the project URL.
* Locally editing an existing CSC level's subtype and confirming that the share/remix experience worked as expected before and after the edit.

It sounds like the curriculum team will own customizing the new levels, so these should work fine initially. Adding a warning about correct use of subtypes in a template-backed progression feels like a good follow-up. Thanks for tracking that already!

File: dashboard/app/views/levels/editors/fields/_special_level_types.html.haml (hunk 1 above)
Comment (molly-moen) [reply dashboard/app/views/levels/editors/fields/_special_level_types.html.haml:62]: That doesn't feel strictly necessary to me, do you think curriculum can generally guess that if they want a new option they should reach out?

Comment (mikeharv) [reply dashboard/app/views/levels/editors/fields/_special_level_types.html.haml:62]: Sure, that's reasonable!

**Pull request merged**